core/src/main/java/org/mskcc/cbio/portal/scripts/MutationFilter.java

```diff
@@ -109,53 +110,57 @@ public boolean acceptMutation(ExtendedMutation mutation) {
           mutationStatusNoneRejects++;
           return false;
       }
-            
-      // Do not accept Silent or Intronic Mutations
-      if( safeStringTest( mutation.getMutationType(), "Silent" ) ||
-               safeStringTest( mutation.getMutationType(), "Intron" ) ){
-         silentOrIntronRejects++;
-         return false;
-      }
-
+      
       // Do not accept LOH or Wildtype Mutations
       if( safeStringTest( mutation.getMutationStatus(), "LOH" ) ||
                safeStringTest( mutation.getMutationStatus(), "Wildtype" ) ){
          lohOrWildTypeRejects++;
          return false;
       }
-
-		// Do not accept Redacted mutations
-		if (safeStringTest(mutation.getValidationStatus(), "Redacted"))
-		{
-		   redactedRejects++;
-		   return false;
-		}
-
-      // Do not accept 3'UTR or 5' UTR Mutations
-      if( safeStringTest( mutation.getMutationType(), "3'UTR" ) ||
-		  safeStringTest( mutation.getMutationType(), "3'Flank" ) ||
-		  safeStringTest( mutation.getMutationType(), "5'UTR" ) ){
-		  utrRejects++;
-         return false;
-      }
       
-      if( safeStringTest( mutation.getMutationType(), "5'Flank" ) ) { 
-            if (whiteListGenesForPromoterMutations.contains(mutation.getEntrezGeneId())){
-                  mutation.setProteinChange("Promoter");
-            } else {
-		  utrRejects++;
-                  return false;
-            }
+      // Do not accept Redacted mutations
+      if (safeStringTest(mutation.getValidationStatus(), "Redacted")) {
+          redactedRejects++;
+          return false;
       }
-
-      // Do not accept IGR Mutations
-      if( safeStringTest( mutation.getMutationType(), "IGR" ) ){
-		  igrRejects++;
-         return false;
+      
+      //Filter by types if specified in the meta file, else filter for the default types
+      if (filteredMutations != null) {
+          Set<String> variantFilteredMutations = new HashSet<String>();
```

Comment:
please make this global and initialize it (i.e. the loop on lines 130-133) in constructor of this class.